components/num-util/src/fixed_decimal.rs

```diff
@@ -699,32 +759,77 @@ fn test_lower_magnitude_bounds() {
     let dec_backup = dec.clone();
     assert_eq!(Error::Limit, dec.multiply_pow10(-1).unwrap_err());
     assert_eq!(dec, dec_backup, "Value should be unchanged on failure");
-}
 
-#[test]
-fn test_from_str_isize_limits() {
-    let dec_max: FixedDecimal = std::isize::MAX.into();
-    let input = dec_max.to_string();
-    let dec: FixedDecimal = FixedDecimal::from_str(&input).unwrap();
-    assert_eq!(dec.to_string(), input);
-    assert_eq!(dec.write_len(), input.len());
-    let dec_min: FixedDecimal = std::isize::MIN.into();
-    let input = dec_min.to_string();
+    // Checking from_str for dec (which is valid)
+    let input = dec.to_string();
     let dec: FixedDecimal = FixedDecimal::from_str(&input).unwrap();
     assert_eq!(dec.to_string(), input);
     assert_eq!(dec.write_len(), input.len());
-}
 
-#[test]
-fn test_from_str_u128_limits() {
-    let dec_max: FixedDecimal = std::u128::MAX.into();
-    let input = dec_max.to_string();
-    let dec: FixedDecimal = FixedDecimal::from_str(&input).unwrap();
-    assert_eq!(dec.to_string(), input);
-    assert_eq!(dec.write_len(), input.len());
-    let dec_min: FixedDecimal = std::u128::MIN.into();
-    let input = dec_min.to_string();
-    let dec: FixedDecimal = FixedDecimal::from_str(&input).unwrap();
-    assert_eq!(dec.to_string(), input);
-    assert_eq!(dec.write_len(), input.len());
+    // The following tests are for testing strings that have sligthly less, sligthtly more, or exactly the extreme
```

Comment:
Split the tests from here down into their own test function; for example, `test_zero_str_bounds`.

There isn't a perfect science on when to make new test functions.  Some people like dozens of tiny test functions with exactly one assert in each.  I tend to go somewhere in the middle, putting closely related tests into the same function, and splitting test functions if they get too unwieldy.